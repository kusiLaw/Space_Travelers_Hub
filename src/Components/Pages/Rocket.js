import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { getRocketData, reserveRocket, cancelReservation } from '../../Redux/Rockets/rocket';
import style from '../style/Rocket.module.css';

const Rocket = () => {
  // get data from store
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets);

  useEffect(() => {
    if (rockets.rocketList.length === 0) {
      dispatch(getRocketData());
    }
  }, []);// call when page load

  return (
    <section className="rocket-container">
      <ul className={style.rocket_list}>
        {
     rockets.rocketList.map((items) => (
       <li key={items.id} className={style.rocket_list_item}>
         <div className="rocket-img-container">
           <img src={items.flickerImages[0]} alt="rocket" width="300" height="280" />
         </div>
         <div className={style.rocket_body_container}>
           <h2>{items.rocketName}</h2>
           <p>
             { items.reserved ? <span className={style.reserved}>Reserved </span> : <></>}
             {items.description}
           </p>
           { items.reserved ? (

             <button
               className={style.btn_cancel}
               type="button"
               onClick={() => {
                 dispatch(cancelReservation(items.id));
               }}
             >
               Cancel Reservation
             </button>
           ) : (
             <button
               className={style.btn_reserved}
               type="button"
               onClick={() => {
                 dispatch(reserveRocket(items.id));
               }}
             >
               Reserve Rocket
             </button>
           )}
         </div>
       </li>
     ))
    }
      </ul>
    </section>
  );
};

export default Rocket;
