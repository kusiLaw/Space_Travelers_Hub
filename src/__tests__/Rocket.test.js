/* eslint-disable no-undef */
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../Redux/configureStore';
import Rocket from '../Components/Pages/Rocket';

it('Testing rockets component, should match snapshot:', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <Rocket />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
