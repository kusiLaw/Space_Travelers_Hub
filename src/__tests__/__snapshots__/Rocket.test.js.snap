// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rockets component, should match snapshot: 1`] = `
<section
  className="rocket-container"
>
  <ul
    className="rocket_list"
  />
</section>
`;
